### PR TITLE
ios_cmake_3_14

### PR DESCRIPTION
**Pull request by crazyhappygame**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,11 +18,11 @@ matrix:
       osx_image: xcode9.3
       sudo: required
     # iOS_cmake
-    # - os: osx
-    #   env: BUILD_TARGET=ios_cmake
-    #   language: cpp
-    #   osx_image: xcode9.3
-    #   sudo: required
+    - os: osx
```
Comment (crazyhappygame) [inline .travis.yml:21]: @minggo It looks like ios_cmake build fails for v4. Please check travis build:
```
/Users/travis/build/cocos2d/cocos2d-x/cocos/renderer/backend/metal/DeviceMTL.h:29:9: fatal error: 'QuartzCore/CAMetalLayer.h' file not found

#import <QuartzCore/CAMetalLayer.h>

        ^~~~~~~~~~~~~~~~~~~~~~~~~~~

1 error generated.
```

**Comment by crazyhappygame** [comment]
@minggo could you merge it?

**Comment by minggo** [comment]
But there is not problems with other PR.

**Comment by crazyhappygame** [comment]
I am so sorry I do not understand...
@minggo this is for https://github.com/cocos2d/cocos2d-x/pull/20040
My plan is following:
Because this is not easy to update cmake for Linux and Android I think that we should update it for now only for mac/ios.
And use for iOS latest cmake.

**Comment by crazyhappygame** [comment]
@minggo why for v4 BUILD_TARGET=ios_cmake Travis build is disabled? I tried to enable it but there is compilation error

**Comment by minggo** [comment]
Can not build iOS for simulator, as simulator doesn't support metal. And account information is needed to build for devices.

**Comment by crazyhappygame** [comment]
According to https://github.com/cocos2d/cocos2d-x/pull/20040 it should be possible to build iOS simulator with latest cmake

**Comment by minggo** [comment]
I don't think so, https://github.com/cocos2d/cocos2d-x/pull/20040 just creates projects for simulator, it can not solve the metal issue. They are different issues. Simulator support metal should wait newer Xcode and macOS version.

**Comment by crazyhappygame** [comment]
Please check 
https://docs.travis-ci.com/user/languages/objective-c/
There Xcode 11 is available for Travis configuration

**Comment by minggo** [comment]
macOS 15+ and Xcode 11+ is needed: https://developer.apple.com/documentation/metal/supporting_simulator_in_a_metal_app?language=objc

**Comment by crazyhappygame** [comment]
@minggo could you merge now?
Changes:
1. IOS cmake build enabled and passed
2. Update cmake to latest version

**Pull request merged**